### PR TITLE
2223: availability optimization

### PR DESCRIPTION
**Pull request by cableman**
This is an optimization of availability from the #1075 pull request.

See https://platform.dandigbib.org/issues/2223

### CODE REVIEW

File: modules/fbs/includes/fbs.availability.inc [hunk 1]
```diff
@@ -94,11 +140,7 @@ function fbs_availability_holdings($provider_ids) {
         $result_holding['placement'][] = $material_description[0]->materialGroupName;
       }
 
-      module_load_include('client.inc', 'opensearch');
-      $entity_id = ding_provider_build_entity_id($provider_ids[0]);
-      $suffix_type = variable_get('fbs_holdings_suffix_type', 'shelf_mark');
```
Comment (kasperg) [inline modules/fbs/includes/fbs.availability.inc]: Moving this *out* of the loop started on L75 seems to be a key change as `ding_provider_build_entity_id()` will invoke the search provider and [`opensearch` will issue a search](https://github.com/ding2/ding2/blob/078ff150143dd461b32ad842007716bf65328937/modules/opensearch/includes/opensearch.search.inc#L624).

This change seems unrelated to the rest of the changes here. Could you extract that into a separate PR?

File: modules/fbs/fbs.info.rej [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  dependencies[] = profile2
+  dependencies[] = text
+  dependencies[] = virtual_field
+--- 14,19 ----
+  dependencies[] = libraries
+  dependencies[] = list
+  dependencies[] = options
+  dependencies[] = profile2
+  dependencies[] = text
+  dependencies[] = virtual_field
```
Comment (kasperg) [inline modules/fbs/fbs.info.rej]: Bemærk at denne fil ikke bør committes.

File: modules/fbs/fbs.info.orig [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+files[] = includes/classes/FBSCacheDrupal.php
+files[] = includes/classes/FBSCacheInterface.php
+files[] = includes/classes/FBSCacheMemory.php
+files[] = includes/classes/FBSDrupalHttpClient.php
+files[] = includes/classes/FBSFakeHttpClient.php
+files[] = includes/classes/FBSGuzzleHttpClient.php
+files[] = includes/classes/FBSLogDrupal.php
+files[] = includes/classes/FBSLogInterface.php
+files[] = includes/classes/FBSLogStdout.php
+files[] = includes/classes/FBSMisconfiguredException.php
```
Comment (kasperg) [inline modules/fbs/fbs.info.orig]: Bemærk at denne fil ikke bør committes.

File: modules/fbs/fbs.info.rej (hunk 2 above)
Comment (cableman) [reply modules/fbs/fbs.info.rej]: Ups

File: modules/fbs/fbs.info.orig (hunk 3 above)
Comment (cableman) [reply modules/fbs/fbs.info.orig]: Ups

File: modules/fbs/includes/fbs.availability.inc (hunk 1 above)
Comment (cableman) [reply modules/fbs/includes/fbs.availability.inc]: No, it's not moved outside the loop???

It checks if the entity id exists in the current loop and loads ding_provider_build_entity_id(reset($item->recordId))

**Comment by guddo** [comment]
I guess `modules/ding_availability/ding_availability.field.inc.orig` and `modules/fbs/includes/fbs.availability.inc.orig` should also not be committed..

**Comment by DDBdrift** [comment]
https://platform.dandigbib.org/issues/2223 | Resolved (tag version)

**Pull request closed without merging**